src/Filters/FederalState.jsx

```diff
@@ -0,0 +1,131 @@
+// It uses this name rather than 'State' to distinguish from the React state
+// Idea https://en.wikipedia.org/wiki/U.S._state
+
+import React from 'react'
+import { connect } from 'react-redux'
+import AggregationItem from './AggregationItem'
+import CollapsibleFilter from './CollapsibleFilter'
+import Typeahead from '../Typeahead'
+import { addMultipleFilters } from '../actions/filter'
+import { THESE_UNITED_STATES } from '../constants'
+import { normalize } from './utils'
+
+const buildLabel = x => {
+  return THESE_UNITED_STATES[x] + ' (' + x + ')'
+}
+
+export class FederalState extends React.Component {
+  constructor(props) {
+    super(props)
+    this._onInputChange = this._onInputChange.bind(this)
+    this._onOptionSelected = this._onOptionSelected.bind(this)
+  }
+
+  render() {
+    return (
+      <CollapsibleFilter title='State'
+                         desc='The state of the mailing address provided by the consumer'
+                         showChildren={this.props.showChildren}
+                         className='aggregation'>
+        <Typeahead placeholder='Enter state name or abbreviation'
+                   onInputChange={this._onInputChange}
+                   onOptionSelected={this._onOptionSelected}
+                   renderOption={this._renderOption} />
+        <ul>
+        {
+          this.props.options.map(bucket => 
+            <AggregationItem item={bucket}
+                             key={bucket.key}
+                             fieldName="state"
+            />
+          )
+        }
+        </ul>
+      </CollapsibleFilter>
+    )
+  }
+
+  // --------------------------------------------------------------------------
+  // Typeahead Helpers
+
+  _onInputChange(value) {
```

Comment:
Should we think about making these methods part of the TypeAhead component so we don't have as much duplicate code in the actual component to test? 

Obviously the `mapStateToProps` and `mapDispatchToProps` will require different helpers as you show below with `buildLabel` but that could be something we take from utils as well.

(As this is an incomplete version feel free to disregard if that's the plan 😄 )